templates/stamp/utilities.sh

```diff
@@ -349,17 +445,397 @@ is_valid_arg()
+        echo "Not all required backup configuration parameters have been set"
+        exit 1;
+    fi
+
+    # persist the settings
+    tee "${backup_configuration}" > /dev/null <<EOF
+BACKUP_STORAGEACCOUNT_NAME={BACKUP_STORAGEACCOUNT_NAME}
+BACKUP_STORAGEACCOUNT_KEY={BACKUP_STORAGEACCOUNT_KEY}
+BACKUP_RETENTIONDAYS={BACKUP_RETENTIONDAYS}
+MONGO_REPLICASET_CONNECTIONSTRING={MONGO_REPLICASET_CONNECTIONSTRING}
+MYSQL_SERVER_LIST={MYSQL_SERVER_LIST}
+DATABASE_USER={DATABASE_USER}
+DATABASE_PASSWORD={DATABASE_PASSWORD}
+TEMP_DATABASE_USER={TEMP_DATABASE_USER}
+TEMP_DATABASE_PASSWORD={TEMP_DATABASE_PASSWORD}
+DATABASE_TYPE={DATABASE_TYPE}
+EOF
+
+    # replace the place holders (using # since the repo path will have forward slashes)
+    log "Populating backup configuration file with appropriate values"
+    sed -i "s#{BACKUP_STORAGEACCOUNT_NAME}#${account_name}#I" $backup_configuration
+    sed -i "s#{BACKUP_STORAGEACCOUNT_KEY}#${account_key}#I" $backup_configuration
+    sed -i "s#{BACKUP_RETENTIONDAYS}#${backupRententionDays}#I" $backup_configuration
+    sed -i "s#{MONGO_REPLICASET_CONNECTIONSTRING}#${mongoReplicaSetConnectionString}#I" $backup_configuration
+    sed -i "s#{MYSQL_SERVER_LIST}#${mysqlServerList}#I" $backup_configuration
+    sed -i "s#{DATABASE_USER}#${databaseUser}#I" $backup_configuration
+    sed -i "s#{DATABASE_PASSWORD}#${databasePassword}#I" $backup_configuration
+    sed -i "s#{TEMP_DATABASE_USER}#${tempDatabaseUser}#I" $backup_configuration
+    sed -i "s#{TEMP_DATABASE_PASSWORD}#${tempDatabasePassword}#I" $backup_configuration
+    sed -i "s#{DATABASE_TYPE}#${databaseType}#I" $backup_configuration
+
+    # this file contains secrets (storage account key). Secure it
+    chmod 600 $backup_configuration
+
+    # create the cron job
+    cron_installer_script="${backup_script}.${databaseType}"
+    lock_file="${cron_installer_script}.lock"
+    install_command="sudo flock -n ${lock_file} bash ${backup_script} -s ${backup_configuration} >> ${backup_log} 2>&1"
+    echo $install_command > $cron_installer_script
+
+    # secure the file and make it executable
+    chmod 700 $cron_installer_script
+
+    # Remove the task if it is already setup
+    log "Uninstalling existing backup job for the '${databaseType}' database(s)"
+    crontab -l | grep -v "sudo bash ${cron_installer_script}" | crontab -
+
+    # Setup the background job
+    log "Installing backup job for the '${databaseType}' database(s)"
+    crontab -l | { cat; echo "${backupFrequency} sudo bash ${cron_installer_script}"; } | crontab -
+    exit_on_error "Failed setting up '${databaseType}' backups." $ERROR_CRONTAB_FAILED
+
+    # setup the cron job
+    log "Completed setting up database backup for '${databaseType}' database(s)"
+    # exit 0;
+}
+
+#############################################################################
+# Set server Time Zone
+#############################################################################
+
+set_timezone()
```

Comment:
It should not. you will notice that the function allows for an override. So the actual region can be changed.